### ISSUE TITLE
How to manage channels and real time updates in this setup? 

### ISSUE THREAD

**Issue by kshitizshankar** [stalniy/casl-feathersjs-example#11]
Is there any example on how we can manage channels and real time updates in this setup? 

With the generated code in feathersjs setup, everyone logged in gets all the updates. Building on top of that, I tried setting up different channels based on resource names and roles but it quickly got out of hands with too much hard coded stuff. 

Is there any base/example that would be helpful in understanding how we can setup channels and limit the events to clients that have access? 

For example, any time a new blog post is created, how can I limit the event to users who have access? (Say Blog is in draft mode and should only be accessible by admins, and I have casl rules that define that - how to use those rules in determining which channels to publish the events to?)

**Comment by stalniy** [comment]
I haven’t worked with feathers for a while, so can’t suggest the solution. But I believe feathers has something like filters for channels. In that filters you need to call `ability.can(...)` and based on the result either send notification or not

**Comment by kshitizshankar** [comment]
I explored that as well but there are a few issues with that approach as it would mean that I have to loop through all the connections and test the ability for each connected user - which won't scale well. 

In order to reduce that, I tried creating individual channels for each controlled resource type (so something like `/organization/org1` and for the access roles of that resource `/organization/org1/admin` `/organization/org1/member` etc.

The issue for me became about figuring out which channels to send which data to. It started getting too complex so I thought there must be an easier way or something that I missed - will keep looking. Thanks a lot!
